src/auto-evo/MicheTree.cs

```diff
@@ -0,0 +1,441 @@
+    /// </summary>
+    private const float DEPTH_SEPARATION = 50.0f;
+
+    /// <summary>
+    ///   Horizontal separation between leafs.
+    /// </summary>
+    private const float LEAF_SEPARATION = 50.0f;
+
+    /// <summary>
+    ///   Each time the control is zoomed, sizeFactor gets multiplied (or divided) by this factor.
+    ///   This allows for smooth zooming, that zoom will not seem too quick when small, nor too slow when big.
+    /// </summary>
+    private const float ZOOM_FACTOR = 0.9f;
+
+    private const float SIZE_FACTOR_MIN = 0.2f;
+    private const float SIZE_FACTOR_MAX = 1.0f;
+
+    /// <summary>
+    ///   Draws outside this margin will be omitted to improve performance.
+    /// </summary>
+    private const float DRAW_MARGIN = 30.0f;
+
+    /// <summary>
+    ///   Default size of <see cref="MicheTreeNode"/>.
+    /// </summary>
+    private static readonly Vector2 TreeNodeSize = new(30, 30);
+
+    /// <summary>
+    ///   Auxiliary vector for <see cref="DRAW_MARGIN"/>
+    /// </summary>
+    private static readonly Vector2 DrawMargin = new(DRAW_MARGIN, DRAW_MARGIN);
+
+    /// <summary>
+    ///   All MicheTreeNodes are in this group so that they work as radio buttons.
+    /// </summary>
+    private readonly ButtonGroup nodesGroup = new();
+
+#pragma warning disable CA2213
+    private Miche? rootMiche;
+    private MicheTreeNode? rootMicheNode;
+
+    /// <summary>
+    ///   Tree part of <see cref="EvolutionaryTree"/>. Consists of many buttons and connection lines.
+    /// </summary>
+    private Control tree = null!;
+
+    // Local copy of fonts
+    private LabelSettings bodyItalicFont = null!;
+    private LabelSettings bodyFont = null!;
+
+    private PackedScene treeNodeScene = null!;
+#pragma warning restore CA2213
+
+    private int treeDepth;
+
+    /// <summary>
+    ///   Drag offset relative to tree.
+    /// </summary>
+    private Vector2 dragOffset;
+
+    private bool dragging;
+    private Vector2 lastMousePosition;
+
+    /// <summary>
+    ///   Default size of lato-small font.
+    /// </summary>
+    private float smallFontSize;
+
+    /// <summary>
+    ///   The tree's size factor.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     <see cref="Control.Scale"/> is not used here so that we know the drawn parts.
+    ///   </para>
+    /// </remarks>
+    private float sizeFactor = 1.0f;
+
+    /// <summary>
+    ///   If redraw is needed.
+    /// </summary>
+    private bool dirty;
+
+    [Signal]
+    public delegate void MicheSelectedEventHandler(int micheHash);
+
+    private Vector2 TreeSize => new(rootMicheNode!.Width + 200, treeDepth * DEPTH_SEPARATION + 100);
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        tree = GetNode<Control>(TreePath);
+
+        treeNodeScene = GD.Load<PackedScene>("res://src/auto-evo/MicheTreeNode.tscn");
+
+        // Font size is adjusted dynamically so this needs to be a copy.
```

Comment:
Is this comment or the code wrong? As far as I can see this is not modified, and `bodyFont` is totally unused. I'll make these changes myself to remove the unused things.